File: Queue/queue.py
class Queue:
    def __init__(self):
        self.queue = []

    def enqueue(self, item):
        self.queue.append(item)

    def dequeue(self):
        if not self.is_empty():
            return self.queue.pop(0)
        else:
            print("Queue is empty. Cannot dequeue.")
            return None

    def peek(self):
        if not self.is_empty():
            return self.queue[0]
        else:
            print("Queue is empty. Cannot peek.")
            return None

    def is_empty(self):
        return len(self.queue) == 0

    def size(self):
        return len(self.queue)


if __name__ == "__main__":
    queue = Queue()

    queue.enqueue(10)
    queue.enqueue(20)
    queue.enqueue(30)

    print("Queue size:", queue.size())
    print("Front of the queue:", queue.peek())

    print("Dequeued item:", queue.dequeue())
    print("Dequeued item:", queue.dequeue())

    print("Is queue empty?", queue.is_empty())

File: Queue/Queue.java
import java.util.LinkedList;

public class Queue {
    private LinkedList<Integer> queueList;

    public Queue() {
        queueList = new LinkedList<>();
    }

    public void enqueue(int item) {
        queueList.add(item);
    }

    public int dequeue() {
        if (!isEmpty()) {
            return queueList.removeFirst();
        } else {
            System.out.println("Queue is empty. Cannot dequeue.");
            return -1; // Return a default value
        }
    }

    public int peek() {
        if (!isEmpty()) {
            return queueList.getFirst();
        } else {
            System.out.println("Queue is empty. Cannot peek.");
            return -1; // Return a default value
        }
    }

    public boolean isEmpty() {
        return queueList.isEmpty();
    }

    public int size() {
        return queueList.size();
    }

    public static void main(String[] args) {
        Queue queue = new Queue();

        queue.enqueue(10);
        queue.enqueue(20);
        queue.enqueue(30);

        System.out.println("Queue size: " + queue.size());
        System.out.println("Front of the queue: " + queue.peek());

        System.out.println("Dequeued item: " + queue.dequeue());
        System.out.println("Dequeued item: " + queue.dequeue());

        System.out.println("Is queue empty? " + queue.isEmpty());
    }
}
